src/scripts/main.js

```diff
@@ -1,3 +1,47 @@
 'use strict';
 
-// write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const tableHeaders = document.querySelectorAll('thead th');
+  const tableBody = document.querySelector('tbody');
+
+  tableHeaders.forEach((header, index) => {
+    header.addEventListener('click', () => {
+      sortElements(index, tableBody);
+    });
+  });
+});
+
+function sortElements(columnIndex, tableBody) {
+  const tableRows = [...tableBody.querySelectorAll('tr')];
+
+  tableRows.sort((rowA, rowB) => {
+    const cellA = rowA.children[columnIndex].innerText.trim();
+    const cellB = rowB.children[columnIndex].innerText.trim();
+
+    let a, b;
+
+    if (!isNaN(cellA.split('$').join('').split(',').join(''))) {
+      a = parseFloat(cellA.split('$').join('').split(',').join(''));
+    } else {
+      a = cellA;
+    }
+
+    if (!isNaN(cellB.split('$').join('').split(',').join(''))) {
+      b = parseFloat(cellB.split('$').join('').split(',').join(''));
+    } else {
+      b = cellB;
+    }
```

Comment:
The code for parsing numbers from cell text is repeated for both `cellA` and `cellB`. Consider refactoring this logic into a separate helper function to avoid duplication and improve readability.